### PR TITLE
Added check for valid bounty path in Actions workflow

### PR DESCRIPTION
**Pull request by benharvie**
- Checks for matching `PackageVulnerabilityID` in the directory name
- Checks for matching `Package.Name` in the directory name
- **WIP**: RegEx check for each to output, specifically, which value is invalid to the user

### CODE REVIEW

File: tools/process-disclosure/contains-a-valid-directory.js [hunk 1]
```diff
@@ -14,22 +14,28 @@ const vulnerabilityJson = await fs
     );
   });
 
+// Get required key values
 const packageId = vulnerabilityJson.PackageVulnerabilityID
 const packageName = vulnerabilityJson.Package.Name
 const packageRegistry = vulnerabilityJson.Package.Registry
```
Comment (adam-nygate) [inline tools/process-disclosure/contains-a-valid-directory.js:20]: Perhaps a good opportunity to use [Object Destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

File: tools/process-disclosure/contains-a-valid-directory.js [hunk 2]
```diff
@@ -14,22 +14,28 @@ const vulnerabilityJson = await fs
...
-const illegalDir = diff.filter((item) => !item.path.test(validDir));
+// Construct valid paths
+const vulnerabilityDir = `bounties/${packageRegistry}/${packageName}/${packageId}/vulnerability.json`
+const readmeDir = `bounties/${packageRegistry}/${packageName}/${packageId}/README.md`
+
+// Filter paths that do not match vulnerability valuess
+const illegalDir = diff.filter((item) => !(item.path == vulnerabilityDir || item.path == readmeDir));
 if (illegalDir.length > 0)
-    core.setFailed("Bounty path does not match vulnerability.json value.");
+    core.setFailed("Bounty path does not match vulnerability.json values.");
```
Comment (adam-nygate) [inline tools/process-disclosure/contains-a-valid-directory.js:29]: Would it not be better to simply compare `process.env.BOUNTY_DIR` and `bounties/${packageRegistry}/${packageName}/${packageId}`?

**Review by adam-nygate — CHANGES_REQUESTED** [review]

File: tools/process-disclosure/contains-a-valid-directory.js (hunk 2 above)
Comment (benharvie) [reply tools/process-disclosure/contains-a-valid-directory.js]: Simplifies things a lot more, implemented, cheers 👻

**Comment by JamieSlome** [comment]
@benharvie - we need the check included in the workflow file - `process-disclosure.yml`.

Cheers! 🍰

File: tools/process-disclosure/contains-a-valid-directory.js (hunk 2 above)
Comment (benharvie) [reply tools/process-disclosure/contains-a-valid-directory.js]: `process.env.BOUNTY_DIR` is not necessarily true, this process will validate this 👻

**Review by JamieSlome — APPROVED** [review]
LGTM! 🍰

**Pull request merged**